File: 백준/Silver/2502. 떡 먹는 호랑이/떡 먹는 호랑이.py
import sys
input = sys.stdin.readline
d, k = map(int, input().split())


def fibo(n):
    if n < 2:
        return(n)
    return(fibo(n-1)+fibo(n-2))

x = fibo(d-2)
y = fibo(d-1)
for a in range(1, 100000):
    if (k-(x*a))%y == 0: 
        print(a, (k-(x*a))//y, sep="\n")
        break

File: 백준/Silver/2502. 떡 먹는 호랑이/떡 먹는 호랑이.js
//문제: 2502
const fs = require("fs");
const filePath = process.platform === "linux" ? "/dev/stdin" : "./input.txt";
let input = fs.readFileSync(filePath).toString().trim().split("\n");

solution(input);
function solution(input) {
  const [a, b] = input[0].split(" ").map(Number);
  const memories = [1, 1];
  for (let i = 2; i <= a; i++) {
    memories.push(memories[i - 2] + memories[i - 1]);
  }
  let a1 = memories[a - 3];
  let b1 = memories[a - 2];
  for (let i = 1; i <= b; i++) {
    let x = (b - a1 * i) / b1;
    if (x > 0 && x == Math.floor(x)) return console.log(`${i}\n${x}`);
  }
}
